lib/yargs-factory.ts

```diff
@@ -2135,6 +2146,19 @@ export class YargsInstance {
           if (this.#exitProcess) setBlocking(true);
           this.showCompletionScript();
           this.exit(0);
+        } else if (
+          this.#figCompletionCommand &&
+          argv._.includes(this.#figCompletionCommand)
+        ) {
+          if (this.#exitProcess) setBlocking(true);
+          this.showFigCompletion();
+          this.exit(0);
+          return this[kPostProcess](
+            argv,
+            populateDoubleDash,
+            !!calledFromCommand,
+            false
+          );
```

Comment:
For reviewers: here I am not sure this is the correct way, I just want to bypass all the middleware/validation logic when the generate fig completion command is run and then exit

Reply:
I changed it in [`31a9467` (#2214)](https://github.com/yargs/yargs/pull/2214/commits/31a9467e036f1d4d7a248bef4b87161cc5a747ed)